docs/architecture/adr-070-unordered-account.md

```diff
@@ -2,214 +2,134 @@
 
-When an un-ordered transaction is included into a block, the transaction hash is
-recorded in a dictionary. New transactions are checked against this dictionary for
-duplicates, and to prevent the dictionary grow indefinitely, the transaction must
-specify `timeout_timestamp` for expiration, so it's safe to removed it from the
-dictionary after it's expired.
+We will introduce new storage of time-based, ephemeral unordered sequences using the SDK's existing KV Store library.
 
-The dictionary can be simply implemented as an in-memory golang map, a preliminary
-analysis shows that the memory consumption won't be too big, for example `32M = 32 * 1024 * 1024`
-can support 1024 blocks where each block contains 1024 unordered transactions. For
-safety, we should limit the range of `timeout_timestamp` to prevent very long expiration,
-and limit the size of the dictionary.
+When an unordered transaction is included in a block, a concatenation of the `timeout_timestamp` and sender’s bech32 address
+will be recorded to state (i.e. `542939323/cosmos1v1234567890AbcDeF`). In cases of multi-party signing, we will use a
+comma-separated list of the addresses that signed the transaction (i.e. `5532231/cosmosv11,cosmosv12,cosmosv13`)
+
+New transactions will be checked against the state to prevent duplicate submissions. To prevent the state from growing indefinitely, we propose the following:
+
+- Define an upper bound for the value of `timeout_timestamp` (i.e. 10 minutes).
+- Extend the PreBlocker method to remove state entries with a `timeout_timestamp` earlier than the current block time.
 
 ### Transaction Format
 
 ```protobuf
 message TxBody {
   ...
-
+          
   bool unordered = 4;
+  google.protobuf.Timestamp timeout_timestamp = 5
 }
 ```
 
 ### Replay Protection
 
-In order to provide replay protection, a user should ensure that the transaction's
-TTL value is relatively short-lived but long enough to provide enough time to be
-included in a block, e.g. ~10 minutes.
-
-We facilitate this by storing the transaction's hash in a durable map, `UnorderedTxManager`,
-to prevent duplicates, i.e. replay attacks. Upon transaction ingress during `CheckTx`,
-we check if the transaction's hash exists in this map or if the TTL value is stale,
-i.e. before the current block time. If so, we reject it. Upon inclusion in a block
-during `DeliverTx`, the transaction's hash is set in the map along with it's TTL
-value.
-
-This map is evaluated at the end of each block, e.g. ABCI `Commit`, and all stale
-transactions, i.e. transactions's TTL value who's now beyond the committed block,
-are purged from the map.
-
-An important point to note is that in theory, it may be possible to submit an unordered
-transaction twice, or multiple times, before the transaction is included in a block.
-However, we'll note a few important layers of protection and mitigation:
-
-* Assuming CometBFT is used as the underlying consensus engine and a non-noop mempool
```

Comment:
this is interesting to me.  If we guarantee that you are using cometbft (which the SDK does), comet just takes care of this for you?

Reply:
did you get this answered?